tsl/src/nodes/decompress_chunk/planner.c

```diff
@@ -410,11 +410,31 @@ decompress_chunk_plan_create(PlannerInfo *root, RelOptInfo *rel, CustomPath *pat
 		foreach (lc, clauses)
 		{
 			RestrictInfo *rinfo = lfirst_node(RestrictInfo, lc);
-			if (is_redundant_derived_clause(rinfo, ipath->indexclauses))
-				continue; /* dup or derived from same EquivalenceClass */
+
+			ListCell *indexclause_cell = NULL;
```

Comment:
NIT: Could you add a comment that this code is based on `is_redundant_derived_clause` but modified for `IndexClause`?  This would make the code easier to read without knowing this change.

Reply:
Added above.